src/server/RequestClient.js

```diff
@@ -31,31 +38,34 @@ module.exports = class RequestClient {
   }
 
   get (path) {
-    return fetch(`${this.hostname}/${path}`, {
+    return fetch(this._getUrl(path), {
       method: 'get',
-      credentials: 'include',
-      headers: {
-        'Accept': 'application/json',
-        'Content-Type': 'application/json'
-      }
+      headers: this.defaultHeaders
     })
       // @todo validate response status before calling json
       .then(this.onReceiveResponse)
       .then((res) => res.json())
   }
 
   post (path, data) {
-    return fetch(`${this.hostname}/${path}`, {
+    return fetch(this._getUrl(path), {
       method: 'post',
-      credentials: 'include',
-      headers: {
-        'Accept': 'application/json',
-        'Content-Type': 'application/json'
-      },
+      headers: this.defaultHeaders,
       body: JSON.stringify(data)
     })
       .then(this.onReceiveResponse)
-      // @todo validate response status before calling json
-      .then((res) => res.json())
+      .then((res) => {
+        if (res.status < 200 || res.status > 300) {
+          throw new Error(res.statusText)
+        }
+        return res.json()
+      })
+  }
+
+  _getUrl (url) {
+    if (url.startsWith('http:') || url.startsWith('https:')) {
```

Comment:
nice refactor :+1: `.startsWith` is not available in IE so a regex is probably preferable

Reply:
@goto-bus-stop fixed